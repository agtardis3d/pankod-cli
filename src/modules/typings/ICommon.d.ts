// #region Local Imports
import { Next2Element, NextElement, SvelteElement, MoleculerElement } from '.';
// #region Local Imports

// TODO: Seperate operations params' types
// TODO: IAnswer and IElement should be kept in their own workbenches.

export declare namespace ICommon {
    export type Project = 'nextjs2' | 'nextjs' | 'svelte' | 'moleculer';
    export type Element = Next2Element &
        NextElement &
        SvelteElement &
        MoleculerElement;

    export interface IAddArgs  {
        args: {
            element: Element;
        }
    }

    export interface ITemplateProps {
        fileName?: string;
        upperFileName?: string;
        isPrivate?: boolean;
        lowerFileName?: string;
        interfaceName?: string;
    }

    export interface IAnswers {
        target?: string;
        isClass?: boolean;
        fileName: string;
        fileType?: string;
        interfaceName?: string;
        classDir?: string;
        upperFileName?: string;
        isPrivate?: boolean;
        hasDatabase?: boolean;
        hasPath?: boolean;
        routePath?: string;
        lowerFileName?: string;
        isPage?: boolean;
        isConnectStore?: boolean;
        hasStyle?: string | boolean;
        pluginType?: string;
        isStyled?: boolean;
        isScss?: boolean;
        isFuncComponent?: boolean;
    }

    export interface IAddIndex {
        dirPath: string;
        getFileContent: Function;
        message: string;
    }

    export interface IAddTest {
        dirPath: string;
        getFileContent: Function;
        message: string;
    }

    export interface IWriteFile {
        dirPath: string;
        getFileContent: Function;
        message: string;
    }

    export interface ICreateTest {
        templatePath: string;
        templateProps: ITemplateProps;
        answers: IAnswers;
        dirPath: string;
        successMessage: string;
    }

    export interface IReplaceContent {
        filetoUpdate: string;
        fileDir: string;
        regexKey: RegExp;
        message: string;
        getFileContent(): string;
    }
}
