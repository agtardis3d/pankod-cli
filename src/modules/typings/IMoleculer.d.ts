// #region Global Imports
import { Question } from "inquirer";
// #region Global Imports

// #region Local Imports
import { ICommon } from ".";
// #region Local Imports

export type MoleculerElement = 'service' | 'repository';

export interface IMoleculerQuestions {
    entity: Question<ICommon.IAnswers>[];
    service: Question<ICommon.IAnswers>[];
    [key: string]: Question<ICommon.IAnswers> | Question<ICommon.IAnswers>[];
}

export interface IMoleculerActions {
    entity: Function;
    service: Function;
    [key: string]: Function;
}

export declare namespace IMoleculerHelper {
	export interface IBrokerHelperTemplatesParams {
		replaceFileDir: string;
		brokerHelperImport: string;
		brokerHelperCreate: string;
	}

	export interface ICreateEntityHelperParams {
		templatePath: string;
		indexTemplate: string;
	}
	export interface ICreateServiceParams {
		templatePath: string;
		indexTemplate: string;
		testTemplatePath: string;
		integrationTemplatePath: string;
		brokerHelperTemplatesParams: IBrokerHelperTemplatesParams;
		createServiceHelperParams: ICreateServiceHelperParams;
		createInterfaceParams: ICreateInterfaceParams;
	}
	export interface ICreateServiceHelperParams {
		templatePath: string;
		indexTemplate: string;
		testTemplatePath: string;
	}

	export interface ICreateRepositoryParams {
		templatePath: string;
		indexTemplate: string;
		testTemplatePath: string;
		createInterfaceParams: ICreateInterfaceParams;
		createEntityTemplatesParams: ICreateEntityHelperParams;
	}

	export interface ICreateInterfaceParams {
		templatePath: string;
		indexInterfaceTemplate: string;
		folderIndexTemplate: string;
	}
}
