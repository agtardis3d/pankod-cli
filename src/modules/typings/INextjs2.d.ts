// #region Global Imports
import {
    QuestionCollection,
    InputQuestion,
    Question,
    ConfirmQuestion,
    ListQuestion
} from 'inquirer';
// #region Global Imports

// #region Local Imports
import { ICommon } from '.';
// #region Local Imports

export type Next2Element = 'Page' | 'ClassComponent' | 'FunctionalComponent';

export interface INextjs2CommonQuestions {
    addStyle: ListQuestion<ICommon.IAnswers>;
    connectStore: ConfirmQuestion<ICommon.IAnswers>;
    enterComponentName: InputQuestion<ICommon.IAnswers>;
    isHaveReducer: ListQuestion<ICommon.IAnswers>;
}

export interface INextjs2Questions {
    Plugin: QuestionCollection<ICommon.IAnswers>;
    ClassComponent: QuestionCollection<ICommon.IAnswers>;
    FunctionalComponent: QuestionCollection<ICommon.IAnswers>;
    Page: QuestionCollection<ICommon.IAnswers>;
    [key: string]: QuestionCollection<ICommon.IAnswers>;
}

export interface INextjs2Actions {
    ClassComponent: Function;
    FunctionalComponent: Function;
    Page: Function;
    // Plugin: Function;
    [key: string]: Function;
}

export declare namespace INextjs2Helper {
    export interface IAddRoutesReplaceParams {
        routesDir: string;
        routesTemplate: string;
    }

    export interface ICreateInterfaceParams {
        templatePath: string;
        pageInterfaceDir: string;
        compInterfaceDir: string;
        pageInterfaceIndex: string;
        compInterfaceIndex: string;
        storeInterface: string;
        storeImportInterface: string;
        interfaceDir: string;
        reduxInterfaceDir: string;
        componentsDir: string;
    }

    export interface ICreateStyle {
        templatePath: string;
        pageDirPath?: string;
        compDirPath: string;
        pageStyledDirPath?: string;
        isStyledComponent?: boolean;
    }

    export interface ICreateClassComponentParams {
        templatePath: string;
        indexTemplatePath: string;
        createInterfaceParams: ICreateInterfaceParams;
        addReducerParams: IAddReducerParams;
        addActionParams: IAddActionParams;
    }

    export interface ICreateFuncComponentParams {
        createInterfaceParams: ICreateInterfaceParams;
        templatePath: string;
        indexTemplatePath: string;
        componentsDir: string;
        componentTestTemplatePath: string;
    }

    export interface IAddReducerParams {
        reducerIndexTemplatePath: string;
        reducerTemplatePath: string;
        reducerStoreTemplatePath: string;
        addActionConstIndexParams: IAddActionConstIndexParams;
        reducerTestTemplatePath: string;
    }

    export interface IAddActionConstIndexParams {
        actionConstTemplatePath: string;
        actionConstsFileDir: string;
    }

    export interface IAddActionParams {
        actionTemplatePath: string;
        actionIndexTemplatePath: string;
        actionTestTemplatePath: string;
    }
}
