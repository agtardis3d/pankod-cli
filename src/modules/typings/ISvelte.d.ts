// #region Global Imports
import { Question } from "inquirer";
// #region Global Imports

// #region Local Imports
import { ICommon } from ".";
// #region Local Imports

export type SvelteElement = 'component';

export interface ISvelteQuestions {
    Component: Question<ICommon.IAnswers>[];
    [key: string]: Question<ICommon.IAnswers> | Question<ICommon.IAnswers>[];
}

export interface ISvelteActions {
    Component: Function;
    [key: string]: Function;
}

export declare namespace ISvelteHelper {
    export interface ICreateStyle {
        templatePath: string;
        compDirPath: string;
    }

    export interface ICreateComponentParams {
        templatePath: string;
        componentsDir: string;
    }
}
