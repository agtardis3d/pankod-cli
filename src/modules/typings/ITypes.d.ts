// #region Global Imports
import { ListQuestion } from "inquirer";
// #region Global Imports

// #region Local Imports
import { ICommon } from ".";
// #region Local Imports

export interface IPankodConfig {
    project: string;
    plugins: string[];
}

export interface IText {
    moleculer: string;
    nextjs: string;
    svelte: string;
    nextjs2: string;
    pankodCli: string;
    [key: string]: string;
}

export interface IQuestions {
    moleculer: ListQuestion<ICommon.IAnswers>;
    nextjs: ListQuestion<ICommon.IAnswers>;
    svelte: ListQuestion<ICommon.IAnswers>;
    nextjs2: ListQuestion<ICommon.IAnswers>;
    [key: string]: ListQuestion<ICommon.IAnswers>;
}

export interface IQuestionsHelper {
    default: {
        showQuestions: Function;
    };
}

export interface IPluginsHelper {
    default: {
        addPlugin: Function;
    };
}

export interface ISupportedCommands {
    moleculer: string[];
    nextjs: string[];
    svelte: string[];
    nextjs2: string[];
    [key: string]: string[];
}
