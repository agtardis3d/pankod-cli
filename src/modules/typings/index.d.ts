export * from './ICommon';
export * from './INextjs2';
export * from './INextjs';
export * from './ISvelte';
export * from './IMoleculer';
export * from './ITypes';
