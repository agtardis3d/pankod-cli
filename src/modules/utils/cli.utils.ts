// #region Local Imports
import * as suitcase from '../henchman/suitcase';
import { ICommon } from '../typings';
import { renderMessage } from '.';
// #endregion Local Imports

// TODO: bind this or what ?
export const validateProject = (project: string) => {
    if (!suitcase.hasOwnProperty(project)) {
        this.error(renderMessage.invalidProject(project));
    }
};

export const validateCommand = (
    element: ICommon.Element,
    project: ICommon.Project
) => {
    if (!suitcase[project].hasOwnProperty(element)) {
        this.error(renderMessage.invalidElement(project, element));
    }
};

export const getAllElements = () => {
    return Object.keys(Object.assign({}, ...Object.values(suitcase)));
};

export const getUsage = () => getAllElements().map(e => `add ${e}`);
