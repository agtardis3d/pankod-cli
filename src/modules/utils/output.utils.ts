// #region Global Imports
import chalk from 'chalk';
// #endregion Global Imports

// TODO: Create a better renderer
export const renderMessage = {
    invalidProject: (project: string) => {
        return `${chalk.bold(project)} is NOT a supported project!`;
    },

    invalidElement: (project: string, element: string) => {
        const [p, e] = [project, element].map(w => chalk.bold(w));

        return chalk.red(
            `${p} projects have no element called ${e} to be created!`
        );
    }
};
