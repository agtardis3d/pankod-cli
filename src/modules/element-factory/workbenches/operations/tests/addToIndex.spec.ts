import * as path from "path"
import * as fs from 'fs';

import { operations } from '../../../../utils';
import { addToIndex } from '../addToIndex.operation';
import { getTemplate } from '../getTemplate.operation';
import { ICommon } from '../../../../typings';

describe(operations, () => {
    describe('addToIndex.operation', () => {
        const params: ICommon.IAddIndex = {
            dirPath: `__temp__/elements/index.ts`,
            getFileContent: () => {
                return getTemplate(
                    path.resolve('./lib/Templates/nextjs/Components/index.mustache'),
                    { fileName: 'Test' }
                );
            },
            message: 'Entity added to index.ts.'
        };

        addToIndex(params);

        const created = String(fs.readFileSync(params.dirPath));

        test('should pass', () => {
            expect(created).toMatchSnapshot();
        });
    });
});
