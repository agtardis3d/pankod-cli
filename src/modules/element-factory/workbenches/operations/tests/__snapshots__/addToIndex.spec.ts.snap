// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[44m[1m[operations][22m[49m -- [3mfs manipulations[23m addToIndex.operation should pass 1`] = `"export { default as Test } from '@Components/Test';export { default as Test } from '@Components/Test';export { default as Test } from '@Components/Test';export { default as Test } from '@Components/Test';export { default as Test } from '@Components/Test';export { default as Test } from '@Components/Test';export { default as Test } from '@Components/Test';export { default as Test } from '@Components/Test';export { default as Test } from '@Components/Test';export { default as Test } from '@Components/Test';export { default as Test } from '@Components/Test';export { default as Test } from '@Components/Test';export { default as Test } from '@Components/Test';export { default as Test } from '@Components/Test';export { default as Test } from '@Components/Test';"`;
